### PR TITLE
Issue #13345: Enable examples tests for IllegalInstantiationCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  IllegalInstantiationCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the ** IllegalInstantiationCheck** in the project. The implementation focuses on ensuring that the  IllegalInstantiation checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by romani** [comment]
Can we copy code of Example3 to Example1 and Example2 to let the be the same by java code ?

**Comment by AmitKumarDeoghoria** [comment]
I was thinking of doing the same. Done.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/Example1.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalInstantiation">
+      <property name="classes"
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/Example1.java:5]: Example1 must be default config.
All other configs are showing change in behavior after customization by properties

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/Example1.java (hunk 1 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/Example1.java:5]: In my previous point I meant to copy example3 Java statements only.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
thanks a lot.

limitation of array is pretty weird, but lets not dig on this now.

**Pull request merged**